src/features/productDetails/ProductDetails.tsx

```diff
@@ -3,15 +3,17 @@ import {Link, useParams} from "react-router-dom";
 import {useQuery} from "@tanstack/react-query";
 import {getProductWithId} from "../../services/api.ts";
 import {useCartStore} from "../cart/cartStore.ts";
+import ProductDetailsLoader from "./ProductDetailsLoader.tsx";
+import AppError from "../error/AppError.tsx";
 
 const ProductDetails = () => {
     const {productId} = useParams()
     const productQuery = useQuery({queryKey: [productId], queryFn: () => getProductWithId(productId ?? '')})
```

Comment:
You can create the hook in a different file for just hooks, then call it and pass the productId